apis/apps/v1alpha1/componentdefinition_types.go

```diff
@@ -454,6 +456,17 @@ type ComponentDefinitionSpec struct {
 	// +optional
 	PodManagementPolicy *appsv1.PodManagementPolicyType `json:"podManagementPolicy,omitempty"`
 
+	// PodUpdatePolicy indicates how pods should be updated
+	//
+	// - `StrictInPlace` indicates that only allows in-place upgrades.
+	// Any attempt to modify other fields will be rejected.
+	// - `PreferInPlace` indicates that we will first attempt an in-place upgrade of the Pod.
+	// If that fails, it will fall back to the ReCreate, where pod will be recreated.
+	// Default value is "PreferInPlace"
+	//
+	// +optional
+	PodUpdatePolicy *v1alpha1.PodUpdatePolicyType `json:"podUpdatePolicy,omitempty"`
```

Comment:
I think this field should be defined in ClusterComponentSpec.

Reply:
And InstanceTemplate

Reply:
@weicao I think this field should be defined in the `ComponentDefinitionSpec`, similar to `PodManagementPolicy`, because I think there is no need for this field to be modified later?

Reply:
> And InstanceTemplate

@weicao It seems there is no scenario where different PodUpdatePolicies need to be specified separately for different InstanceTemplate instances under the same component?

Reply:
> > And InstanceTemplate
> 
> @weicao It seems there is no scenario where different PodUpdatePolicies need to be specified separately for different InstanceTemplate instances under the same component?

Yeah, you're correct